routing/pathfind.go

```
@@ -1237,18 +1241,81 @@ func findBlindedPaths(g Graph, target route.Vertex,
 		return features.HasFeature(lnwire.RouteBlindingOptional), nil
 	}
 
+	// Initialize the map to track visited vertices
+	alreadyVisited := map[route.Vertex]bool{}
+	partiallySpecifiedPath := make([]blindedHop, 0)
+
+	// Sanity checking the partially specified blinded path.
+	// Check if the order makes sense (ie, the chain is valid).
+	// Check if the all the channels exist.
+	// Check if all the nodes involved are advertising route blinding.
+	node := target
+	for _, chanID := range restrictions.incomingChainedChannels {
+		// iterate over the node's channels in search for paths to the
+		// currentTarget
+		err := g.ForEachNodeChannel(node,
+			func(channel *graphdb.DirectedChannel) error {
+				if channel.ChannelID == chanID {
+					_, err := supportsRouteBlinding(
+						channel.OtherNode)
+					if err != nil {
+						return err
+					}
```

should do the check before, if the node does not support we fail fast.